### ISSUE TITLE
Token::equals also compares line numbers

### ISSUE THREAD

**Issue by mbeccati** [FriendsOfPHP/PHP-CS-Fixer#971]
Comparison between two tokens coming from the parser is always failing, as the line number is checked as well. Unless they are on the same line, of course ;)

```php
        $a = new Token(array(T_OPEN_TAG, '<?php', 1));
        $b = new Token(array(T_OPEN_TAG, '<?php', 10));
        $this->assertTrue($a->equals($b));
```

**Comment by keradus** [comment]
If one pass a line number then it will be compared, if not - then not. It is a feature, and you want to remove it.

**Comment by mbeccati** [comment]
So:

* there are no tests to Token::equals at all to confirming that the behaviour is correct
* the change doesn't break any other test
* equals ends up being used in Tokens::getNextTokenOfKind(), where "kind" certainly doesn't give the idea that line numbers are taken into account.
* such "feature" has been removed in master, due to the change to getPrototype()

I'll be happy to make the necessary changes to follow your guidelines. In case you don't think it's a bug in 1.5, feel free to close this, but maybe it's worth marking the ticket as fixed for 2.0, possibly rewording as "Remove feature" ;)

**Comment by keradus** [comment]
> there are no tests for Token::equals that confirm that the behaviour is correct

One days there were almost no tests. That doesn't change the fact that this is the current behavior that you propose to change and it will break BC!


> the change doesn't break any other test

We promise the interface and we can break it only on next MAJOR release (2.x). Even if your suggestion won't change our tests please remember that one can download the project as vendor one and use the classes for his purposes, or simply create a fixers of his own based on that interface. You want to change public method behavior, and that won't be accepted without bumping the major version.


> equals ends up being used in Tokens::getNextTokenOfKind(), where "kind" certainly doesn't give the idea that line numbers are taken into account.

No. One may use it both ways. Because we allow to use it both ways. And when we use it we don't check the line, see: https://github.com/FriendsOfPHP/PHP-CS-Fixer/search?p=2&q=getNextTokenOfKind&utf8=✓


> such "feature" has been removed in master, due to the change to getPrototype()

Yes, I have already mentioned that in your PR, didn't I? 


> I'll be happy to make the necessary changes to follow your guidelines. In case you don't think it's a bug in 1.5, feel free to close this, but maybe it's worth marking the ticket as fixed for 2.0, possibly rewording as "Remove feature" ;)

No, it is not a bug. It's a flexibility to check whatever one need to check. It is a designed behavior - you have even read it in documentation that you have changed in your PR.
2.0 has a lot of interface changes.
